### ISSUE TITLE
Wave Game interaction model does not work for alt input

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/fourier-making-waves#91]
Here's an interesting a11y design problem...

In the Wave Game, the student's answer to a challenge is not evaluated until they have released all sliders.  So they can't solve the challenge unintentionally by simply moving past the correct amplitude - releasing the slider indicates that they have intentionally selected a value. 

But that model only works for mouse/touch.  For keyboard input, there is nothing similar to "student released the slider".  So the user will solve the challenge as soon as they set the amplitude to the correct value using the keyboard arrow.  And I can imagine a similar problem with any future input device that does not have a "release" action. 

This seems like a big problem, especially in Level 1.

I can't think of a way to resolve this using the current UI model and a11y implementation.  (And by the way... This is another example of the downside of not generalizing user-action in scenery, see https://github.com/phetsims/scenery/issues/1238.  DragListener and KeyboardDragListener have different APIs, with the latter having no "press" and "release".)

Some options:

(1) Add a "Check Answer" button, that the user needs to explicitly click.  This button can be clicked as many times as needed, and will still award points. Other sims only reward points if your first "Check Answer" is successful, but this sim seems different.  This will require a lot more tabbing around on the keyboard, and requires revisiting pdomOrder.  EDIT: This option would also facilitate a more straightforward solution deciding when to enabled the "Show Answer" button, and it would make alt input poroblem #87 a non-issue.

(2) Wait until the student tabs away from a slider to evaluate the challenge.  This appears to be technically possible by listening to PressListener `isFocusedProperty`.  But it's highly unintuitive from a UX perspective. 

(3) Change the a11y Slider implementation to have the same "drag" API for all forms of alt input.  This sounds like a huge change (if it's even possible), definitely involves a big discussion in https://github.com/phetsims/scenery/issues/1238, and is unlikely to happen in the Fourier 1.0 timeframe. 

My recommendation is (1).

@arouinfar @kathy-phet thoughts?

**Comment by pixelzoom** [comment]
Note to self, if we go with option (1) above:
* definitely delete `this.numberOfSlidersDraggingProperty` in WaveGameAmplitudesChartNode
* maybe delete `this.isMatchedProperty` in WaveGameLevel
* revise `faceVisibleProperty` derivation in WaveGameLevelNode
* delete `Property.multilink` on line 321 of WaveGameLevelNode
* add 'Check Answer' button in WaveGameLevelNode, listener should do what WaveGameLevel `this.isMatchedProperty` does
* WaveGameLevel `AMPLITUDE_THRESHOLD` is probably unnecessary

And yes, this is the simple solution :)

**Comment by pixelzoom** [comment]
Zoom with @arouinfar, decisions:

* Add "Check Answer" button, above "Show Answer" button
* Add hot key for "Check Answer" button
* After solving challenge or pressing "Show Answer", disable "Check Answer" button
* Press "Check Answer" button 2x to enable "Show Answer"
* "Check Answer" is disabled until the user changes 1 amplitude.
* When challenge is in "explore" mode (after solving or showing answer), it works like it does now. Answer is evaluated when all sliders are release. Answer is evaluated as keyboard up/down arrow is pressed.

**Comment by pixelzoom** [comment]
Everything is implemented, but the hot key (Alt-C) is not working. There are two problems in a11y common code that need to be resolved: https://github.com/phetsims/scenery/issues/1241, https://github.com/phetsims/scenery/issues/1053.

In the meantime, @arouinfar please review everything except the hot key.

**Comment by pixelzoom** [comment]
@arouinfar The hotkey for "Check Answer" is now working.  It's Alt+C, let me know if you'd like to use something else. 

Adding the hotkey to the keyboard help dialog will be done in #93.

**Comment by pixelzoom** [comment]
@arouinfar Wave Game review was completed (and closed) in https://github.com/phetsims/fourier-making-waves/issues/69, and presumably included the features in this issue.  So can this issue be closed?

**Comment by arouinfar** [comment]
Yep, let's close this one.
